Test/Hilbert/test_hilbert.py

```diff
@@ -102,18 +102,21 @@ def test_random_states():
         print("name", name)
         if hi.is_discrete:
             rstate = np.zeros(hi.size)
-
             local_states = hi.local_states
-
             for i in range(100):
-                hi.random_state(rstate)
+                hi.random_state(out=rstate)
                 for state in rstate:
                     assert state in local_states
 
+            assert hi.random_state().shape == (hi.size,)
+            assert hi.random_state(10).shape == (10, hi.size)
+            assert hi.random_state(size=10).shape == (10, hi.size)
+            assert hi.random_state(size=(10,)).shape == (10, hi.size)
+            assert hi.random_state(size=(10, 2)).shape == (10, 2, hi.size)
```

Comment:
why this is not (hi.size,10,2) ?

Reply:
Because batch dimensions come first. Otherwise, it could not be consistent with `size=10`...